independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/BeanManagerImpl.java

```diff
@@ -226,6 +228,11 @@ public Context getContext(Class<? extends Annotation> scopeType) {
         return context;
     }
 
+    @Override
+    public Collection<Context> getContexts(Class<? extends Annotation> scopeType) {
+        return Collections.unmodifiableList(Arc.container().getContexts(scopeType));
```

Comment:
I believe that `Arc.container().getContexts(scopeType)` already returns an immutable list, although it's not mentioned in the javadoc. So maybe just update the javadoc and skip the `Collections.unmodifiableList()` here.

Reply:
OK, I didn't dive too much into this previously, but you're right -- `io.quarkus.arc.impl.Contexts` always returns immutable lists. I'll change this.

Reply:
Done.